Tests/Unit/Microsoft365DSC/Microsoft365DSC.EXORoleGroup.Tests.ps1

```diff
@@ -0,0 +1,175 @@
+        -ChildPath '\Stubs\Generic.psm1' `
+        -Resolve)
+Import-Module -Name (Join-Path -Path $M365DSCTestFolder `
+        -ChildPath '\UnitTestHelper.psm1' `
+        -Resolve)
+
+$Global:DscHelper = New-M365DscUnitTestHelper -StubModule $CmdletModule `
+    -DscResource 'EXORoleGroup' -GenericStubModule $GenericStubPath
+Describe -Name $Global:DscHelper.DescribeHeader -Fixture {
+    InModuleScope -ModuleName $Global:DscHelper.ModuleName -ScriptBlock {
+        Invoke-Command -ScriptBlock $Global:DscHelper.InitializeScript -NoNewScope
+
+        BeforeAll {
+            $secpasswd = ConvertTo-SecureString 'test@password1' -AsPlainText -Force
+            $Credential = New-Object System.Management.Automation.PSCredential ('tenantadmin@mydomain.com', $secpasswd)
+
+            $Global:PartialExportFileName = 'c:\TestPath'
+
+            Mock -CommandName Save-M365DSCPartialExport -MockWith {
+            }
+
+            Mock -CommandName New-M365DSCConnection -MockWith {
+                return 'Credentials'
+            }
+
+            Mock -CommandName Get-PSSession -MockWith {
+            }
+
+            Mock -CommandName Remove-PSSession -MockWith {
+            }
+
+            # Mock Write-Host to hide output during the tests
+            Mock -CommandName Write-Host -MockWith {
+            }
+            Function get-RoleGroup{
+                Return @{
+                    Name        = 'Contoso Role Group'
+                    Members     = 'Exchange Administrator'
+                    Roles       = 'Address Lists'
+                    Description = 'This is the Contoso Role Group'
+                }
+            }
+
+            Function New-RoleGroup{
+                Return $null
+            }
+
+            Function Remove-RoleGroup{
+                Return $null
+            }
+
+            Function Get-RoleGroupMember{
+                Return $null
+            }
+        }
+
+        # Test contexts
+        Context -Name 'Role Group should exist. Role Group is missing. Test should fail.' -Fixture {
+            BeforeAll {
+                $testParams = @{
+                    Name        = 'Contoso Role Group'
+                    Members     = 'Exchange Administrator'
+                    Roles       = 'Address Lists'
+                    Description = 'This is the Contoso Role Group'
+                    Ensure      = 'Present'
+                    Credential  = $Credential
+                }
+
+            }
+
+            It 'Should return false from the Test method' {
+                Mock -CommandName Get-RoleGroup -MockWith {
+                    return $null
+                }
+                Test-TargetResource @testParams | Should -Be $false
+            }
+
+            It 'Should call the Set method' {
+                Mock -CommandName New-RoleGroup -MockWith {
+                    return $null
+                }
+                Set-TargetResource @testParams
```

Comment:
I believe you should check for something here, like Should -Invoke New-RoleGroup. Without adding a check for something specific, it is possible that the code returns something totally different and still succeed.

More info: https://pester.dev/docs/commands/Should#example-1